src/tests/test_engine.py

```diff
@@ -327,32 +329,33 @@ class TestAmoebaWater6:
     Multipole moments: < 0.001 Debye / Debye Angstrom
     Multipole moments (optimized): < 0.01 Debye / Debye Angstrom
     """
-    @classmethod
-    def setup_class(cls):
+    def setup_method(self, method):
+        super().setup_method(method)
         #self.logger.debug("\nBuilding options for target...\n")
-        cls.cwd = os.path.dirname(os.path.realpath(__file__))
-        os.chdir(os.path.join(cls.cwd, "files", "amoeba_h2o6"))
-        if not os.path.exists("temp"):
-            os.makedirs("temp")
-        os.chdir("temp")
+        self.cwd = os.path.dirname(os.path.realpath(__file__))
+        os.chdir(os.path.join(self.cwd, "files", "amoeba_h2o6"))
+        tmpfolder = os.path.join(self.cwd, "files", "amoeba_h2o6", "temp")
+        if not os.path.exists(tmpfolder):
+            os.makedirs(tmpfolder)
+        os.chdir(tmpfolder)
         os.system("ln -s ../prism.pdb")
         os.system("ln -s ../prism.key")
         os.system("ln -s ../hex.arc")
         os.system("ln -s ../water.prm")
         os.system("ln -s ../amoebawater.xml")
-        cls.O = OpenMM(coords="hex.arc", pdb="prism.pdb", ffxml="amoebawater.xml", precision="double", \
+        self.O = OpenMM(coords="hex.arc", pdb="prism.pdb", ffxml="amoebawater.xml", precision="double", \
                             mmopts={'rigidWater':False, 'mutualInducedTargetEpsilon':1e-6})
         tinkerpath = which('testgrad')
         if tinkerpath:
-            cls.T = TINKER(coords="hex.arc", tinker_key="prism.key", tinkerpath=tinkerpath)
+            self.T = TINKER(coords="hex.arc", tinker_key="prism.key", tinkerpath=tinkerpath)
 
     @classmethod
-    def teardown_class(cls):
+    def teardown_class(self):
```

Comment:
Let's take a close look at these and determine which lines should run in test_class and which should run in test_method.